cmd/soroban-cli/src/events.rs

```diff
@@ -0,0 +1,191 @@
+use std::io;
+use std::io::Write;
+
+use clap::Parser;
+use hex::FromHexError;
+use soroban_env_host::xdr::{ReadXdr, ScVal};
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+use crate::rpc::Event;
+use crate::utils;
+use crate::{rpc, rpc::Client};
+use crate::{HEADING_RPC, HEADING_SANDBOX};
+
+#[derive(Parser, Debug)]
+#[clap()]
+pub struct Cmd {
+    /// The ledger range to pull events from
+    start_ledger: u32,
+    end_ledger: u32,
+
+    /// RPC server endpoint
+    #[clap(long,
+        env = "SOROBAN_RPC_URL",
+        help_heading = HEADING_RPC,
+        conflicts_with = "sandbox",
+    )]
+    rpc_url: Option<String>,
+
+    /// Sandbox file from which to pull events from
+    #[clap(
+        long,
+        parse(from_os_str),
+        value_name = "PATH",
+        env = "SOROBAN_LEDGER_FILE",
+        help_heading = HEADING_SANDBOX,
+        conflicts_with = "rpc-url",
+    )]
+    ledger_file: Option<std::path::PathBuf>,
+
+    /// A set of (up to 5) contract IDs to filter events on
+    #[clap(long = "ids", multiple = true)]
+    contract_ids: Vec<String>,
+
+    /// A set of (up to 5) topic filters to filter events on
+    #[clap(long, multiple = true)]
+    topics: Vec<String>,
+
+    /// Formatting options for outputted events
+    #[clap(
+        long,
+        possible_values = ["console", "json"],
+        default_value = "console"
+    )]
+    format: String,
```

Comment:
:bulb: Fyi you can use an `enum` for this field type, which will mean elsewhere in the code you can access the value as an enum rather than as a string. It will ensure everywhere we use the enum we exhaustively consider all paths, etc.

Here's an example for how to do it:
https://github.com/stellar/soroban-tools/blob/e4d3ed4d85181eabf745681b69b039aad028cb8f/cmd/soroban-cli/src/gen.rs#L15-L25

Reply:
Good idea! String comparisons are silly :smile:

Reply:
`r#output` seems like magic :magic_wand:

Reply:
Oh, you don't need the `r#`. If you have a identifier that is the same word as a keyword, you can put `r#` on the front of it to force it to be an identifier. You shouldn't need to use it in your case.